Formula/c/chapel.rb

```diff
@@ -31,6 +32,16 @@ def llvm
     deps.map(&:to_formula).find { |f| f.name.match? "^llvm" }
   end
 
+  # determine the C backend to use based on the system
+  def cbackend
+    on_macos do
+      return "clang"
+    end
+    on_linux do
+      return "gnu"
+    end
+  end
```

Comment:
this can be simplee as 


```suggestion
  cbackend = OS.mac? ? "clang" : "gnu"
```

in the install block?

Reply:
I'll also note, `cbackend` is needed in both `install` and `test`, which is why its a toplevel function. I can put it in `install`, but it will also need to be duplicated in `test`

Reply:
Addressed in https://github.com/Homebrew/homebrew-core/pull/247073